pkg/gateway/server/oauth_apps.go

```diff
@@ -445,6 +446,30 @@ func (s *Server) callbackOAuthApp(apiContext api.Context) error {
 			CreatedAt:    time.Now(),
 			RefreshToken: googleTokenResp.RefreshToken,
 		}
+	case types2.OAuthAppTypeSalesforce:
+		salesforceTokenResp := new(types.SalesforceOAuthTokenResponse)
+		if err := json.NewDecoder(resp.Body).Decode(salesforceTokenResp); err != nil {
+			return fmt.Errorf("failed to parse token response: %w", err)
+		}
+		issuedAt, err := strconv.ParseInt(salesforceTokenResp.IssuedAt, 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse token response: %w", err)
+		}
+		createdAt := time.Unix(issuedAt/1000, (issuedAt%1000)*1000000)
+
+		tokenResp = &types.OAuthTokenResponse{
+			State:        state,
+			TokenType:    salesforceTokenResp.TokenType,
+			Scope:        salesforceTokenResp.Scope,
+			AccessToken:  salesforceTokenResp.AccessToken,
+			ExpiresIn:    7200, // Relies on Salesforce admin not overriding the default 2 hours
+			Ok:           true, // Assuming true if no error is present
+			CreatedAt:    createdAt,
+			RefreshToken: salesforceTokenResp.RefreshToken,
+			Extras: map[string]string{
+				"GPTSCRIPT_SALESFORCE_URL": salesforceTokenResp.InstanceURL,
```

Comment:
This approach is definitely more flexible, but it feels like we're losing the self-documenting property of the credentials framework; i.e. we could put the `token` here too and entirely remove the `with GPTSCRIPT_SALESFORCE_TOKEN as token` clause from `salesforce/credential/tool.gpt`

Reply:
The difference between putting the token here and using the credential framework is that a user won't be prompted if the token doesn't exist. Additionally, the credential framework will handle refreshes. So I think the token should stay the way it is.

Another possibility for this `GPTSCRIPT_SALESFORCE_URL` is for it to have its own credential. That is the way environment variables are handled for workflows: each environment variable is turned into a credential on the tool.

However, I'm not sure that is better for this situation. I think I like this approach.

Reply:
> that a user won't be prompted if the token doesn't exist.

That's great point. OK, you've convinced me.